openfl/component/aggregator/aggregator.py

```diff
@@ -1103,11 +1097,11 @@ def _compute_validation_related_task_metrics(self, task_name) -> dict:
             agg_tensor_key = TensorKey(tensor_name, origin, round_number, report, new_tags)
             # Check if secure aggregation is enabled, set aggregation function.
             agg_function = (
-                task_agg_function
-                if "metric" not in tags
+                WeightedAverage()
+                if "metric" in tags
                 else SecureWeightedAverage()
                 if self._secure_aggregation_enabled
```

Comment:
Isn't the goal of secure aggregation to compute a weighted average to unmask the values first? This implies that all collaborator updates are the same. In this case, the standard flow should work. A weighted average of tensors where all tensors are same, is the tensor itself. 

Motivation is to not do a lot of conditional branching (here and everywhere in the code)